webapi/src/main/scala/org/knora/webapi/slice/common/repo/rdf/RdfModel.scala

```diff
@@ -341,6 +341,13 @@ final case class RdfResource(private val res: Resource) {
   def getObjectIris(propertyIri: String): IO[RdfError, Chunk[InternalIri]] =
     getObjectUris(propertyIri).map(_.map(InternalIri))
 
+  def getObjectIrisConvert[A](prop: String)(implicit map: String => Either[String, A]) =
+    getObjectIris(prop).flatMap { iris =>
+      ZIO.foreach(iris) { iri =>
```

Comment:
**question (cosmetic; non-blocking):** I haven't checked it (and I'm always insecure without a compiler to check me), but I think you can replace `a.flatmap{x => ZIO.foreach(x){...}}` with `ZIO.foreachZIO(a){...}`

Reply:
The `ZIO.foreach` would require a pure value whereas `getObjectIris(prop)` is a `ZIO`.

Reply:
I think there is an effectful variant `foreachZIO`